### PR TITLE
Pr corrections/components stateless

### PR DESCRIPTION
**Pull request by pdeleastar**

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -23,6 +23,55 @@
...
+  display: flex;
+  width: 100%;
+}
+
+.sidebar {
+  order: 1;
+  flex: 1;
+  flex-basis: auto;
+  padding: 0.5rem;
+  min-height: 666px;
```
[the remark this answers is not in the document]
Comment (pdeleastar) [reply src/App.css]: Removed px from css

File: src/App.js [hunk 2]
```diff
@@ -1,19 +1,68 @@
...
+        <div>
+          <div>
+            <Input />
+          </div>
+        </div>
+
+        <div className='flexbox-container'>
+          <div className='main'>
+            {this.props.chart && (
+              <div className='chartContainer'>
```
[the remark this answers is not in the document]
Comment (pdeleastar) [reply src/App.js]: App.js cleaned up, less divs now

File: src/actions/actions.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    type: PEER_ACTION,
+    payload
+  }
+}
+
+export function handleSearch (input) {
+  return dispatch => {
+    fetch(`https://api.iextrading.com/1.0/stock/${input}/stats`)
+      .then(res => res.json())
+      .then(json => dispatch(keyStatAction(json)))
```
[the remark this answers is not in the document]
Comment (pdeleastar) [reply src/actions/actions.js]: Loading, ok, ko features for the UI will be added in a future branch

File: src/components/Input.jsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+            onChange={this.handleChange}
+          />
+        </label>
+        <input type="submit" value="Submit" />
+      </form>
+    );
+  }
+}
+
+const mapDispatchToProps = dispatch => {
```
[the remark this answers is not in the document]
Comment (pdeleastar) [reply src/components/Input.jsx]: mapDispatchToProps function removed

File: src/App.js [hunk 5]
```diff
@@ -1,19 +1,68 @@
...
+      <div>
+        <div>
+          <div>
+            <Input />
+          </div>
+        </div>
+
+        <div className='flexbox-container'>
+          <div className='main'>
+            {this.props.chart && (
```
[the remark this answers is not in the document]
Comment (pdeleastar) [reply src/App.js]: loading, ko, ok features will be added in a future branch, conditional logic now moved one level down in to the components themselves

[the remark this answers is not in the document]
Comment (AbrahamAlcaina) [reply src/App.js]: :+1:

File: src/components/Input.jsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (AbrahamAlcaina) [reply src/components/Input.jsx]: :+1:

File: src/actions/actions.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (AbrahamAlcaina) [reply src/actions/actions.js]: :+1:

File: src/App.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (AbrahamAlcaina) [reply src/App.js]: :+1:

File: src/App.css [hunk 6]
```diff
@@ -23,6 +23,55 @@
...
+  flex: 5;
+  padding: 0.5rem;
+  min-width: 70%;
+}
+
+.chartContainer {
+  height: 600px;
+}
+
+.newsList {
```
[the remark this answers is not in the document]
Comment (AbrahamAlcaina) [reply src/App.css]: done!

File: src/App.css (hunk 1 above)
[the remark this answers is not in the document]
Comment (AbrahamAlcaina) [reply src/App.css]: :+1:

File: src/actions/actions.js [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+function storeInputAction (payload) {
+  return {
+    type: SEARCH,
+    payload
+  }
+}
+
+export function handleSearch (input) {
+  return dispatch => {
+    fetch(`https://api.iextrading.com/1.0/stock/${input}/company`)
```
Comment (AbrahamAlcaina) [inline src/actions/actions.js]: I still missing the .`catch`

File: src/actions/actions.js [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+    fetch(`https://api.iextrading.com/1.0/stock/${input}/peers`)
+      .then(res => res.json())
+      .then(json => dispatch(peerAction(json)))
+    fetch(`https://api.iextrading.com/1.0/stock/${input}/quote`)
+      .then(res => res.json())
+      .then(json => dispatch(keyStatAction(json)))
+    fetch(`https://api.iextrading.com/1.0/market`)
+      .then(res => res.json())
+      .then(json => dispatch(marketAction(json)))
+      .then(dispatch(storeInputAction(input)))
```
Comment (AbrahamAlcaina) [inline src/actions/actions.js]: dispatch returns a promise?

File: src/components/ChartFilter.jsx [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+
+class ChartFilter extends React.Component {
+  filter = time => {
+    this.props.chartFilter(this.props.input, time);
+  };
+
+  render() {
+    return (
+      <div>
+        <button onClick={() => this.filter("1d")}>1d</button>
```
Comment (AbrahamAlcaina) [inline src/components/ChartFilter.jsx]: Avoid functions in the render
https://medium.freecodecamp.org/react-pattern-extract-child-components-to-avoid-binding-e3ad8310725e

File: src/reducers/reducer.js [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+  NEWS_ACTION,
+  CHART_ACTION,
+  PEER_ACTION,
+  MARKET_ACTION
+} from '../actions/actions'
+
+export const reducer = (state = {}, action) => {
+  switch (action.type) {
+    case SEARCH:
+      return Object.assign({}, state, {
```
Comment (AbrahamAlcaina) [inline src/reducers/reducer.js]: tip: you can write something like this:
`return {...state, {input: action.payload}}`

Comment (pdeleastar) [reply src/reducers/reducer.js]: This is changed now

File: src/components/ChartFilter.jsx (hunk 9 above)
Comment (pdeleastar) [reply src/components/ChartFilter.jsx]: This is changed, now using: 
onClick={this.filter} value={"1d"}

File: src/actions/actions.js (hunk 8 above)
Comment (pdeleastar) [reply src/actions/actions.js]: action moved to top of handler now

File: src/actions/actions.js (hunk 7 above)
Comment (pdeleastar) [reply src/actions/actions.js]: Added catch with additional Error component which displays on error

**Pull request merged**